kv/txn_coord_sender.go

```diff
@@ -337,7 +337,12 @@ func (tc *TxnCoordSender) Send(ctx context.Context, ba roachpb.BatchRequest) (*r
 				// spans optimization to 1pc transactions.
 				distinctSpans = len(txnMeta.keys) == 0
 			}
-			ba.IntentSpanIterate(func(key, endKey roachpb.Key) {
+			// We can't pass in a batch response here to better limit the key
+			// spans as we don't know what is going to be affected. This will
+			// affect queries such as `DELETE FROM my.table LIMIT 10` when
+			// executed as a 1PC transaction. e.g.: a (BeginTransaction,
+			// DeleteRange, EndTransaction) batch.
```

Comment:
This might be a reason to say that transactions consisting only of DelRange with limits should always go through the 2PC path.